### PR TITLE
nikshay outcome repeater

### PR DESCRIPTION
**Pull request by proteusvacuum**
@NoahCarnahan @mkangia 
No tests, but should be pretty good to go. Putting up here just in case.
(includes commits from https://github.com/dimagi/commcare-hq/pull/14873)

### CODE REVIEW

**Comment by mkangia** [comment]
Thanks @proteusvacuum.

File: custom/enikshay/integrations/nikshay/repeaters.py [hunk 1]
```diff
@@ -35,13 +36,40 @@ def allowed_to_forward(self, episode_case):
...
+        from custom.enikshay.integrations.nikshay.views import NikshayTreatmentOutcomesView
+        return reverse(NikshayTreatmentOutcomesView.urlname, args=[domain])
+
+    def allowed_to_forward(self, episode_case):
+        allowed_case_types_and_users = self._allowed_case_type(episode_case) and self._allowed_user(episode_case)
+        episode_case_properties = episode_case.dynamic_case_properties()
+        return allowed_case_types_and_users and (
+            not episode_case_properties.get('nikshay_registered', 'false') == 'true' and
+            not episode_case_properties.get('nikshay_id', False) and
+            not episode_case_properties.get('treatment_outcome_nikshay_registered', False) and
```
Comment (NoahCarnahan) [inline custom/enikshay/integrations/nikshay/repeaters.py]: Is the `treatment_outcome_nikshay_registered` case property a boolean? If so I think this should be
```python
episode_case_properties.get('treatment_outcome_nikshay_registered', False) == 'true'
```

**Review by NoahCarnahan — COMMENTED** [review]
Looks good to me, just one question.

**Comment by mkangia** [comment]
@NoahCarnahan @proteusvacuum Following up with team for clarifications on APIs.  Started with FollowUp and HIVTest. Will also co-ordinate for Treatment Outcome after that. Better to hold of on merged till then.

**Comment by mkangia** [comment]
I am going to extract out just the changes for Treatment Outcome and ignore other commits((includes commits from #14873)) to make this cleaner and write test cases. Also included change made by @NoahCarnahan for check.
Also @proteusvacuum, retaining `_save_error_message`. I strongly feel its good to retain that at a single place. Making it flexible to incorporate different fields.

**Comment by mkangia** [comment]
Closing this PR and opening a new PR for treatment outcome API integration #15347

**Pull request closed without merging**